Modules/AwardDKP.lua

```diff
@@ -0,0 +1,105 @@
+local _, core = ...;
+local _G = _G;
+local MonDKP = core.MonDKP;
+local L = core.L;
+
+local function GetEligibleGuildMembers(onlineOnly, sameZone, currZone)
+    -- Get list of guild members, optionally filtered for only online members and/or members currently in zone currZone
+    -- Returns table playerTable where playerTable[playerName] is not nil iff playerName is eligible
+    local playerTable = {}
+    for playerIndex = 1, GetNumGuildMembers() do
+        local name, _, _, _, _, zone, _, _, online, _, _, _, _, _, _, _ = GetGuildRosterInfo(playerIndex);
+        if ((not onlineOnly) or online) and ((not sameZone) or (zone == currZone)) then
+            playerTable[name] = 1
+        end
+    end
+    return playerTable
+end
+
+function MonDKP:AwardPlayer(name, amount)
+	local search = MonDKP:Table_Search(MonDKP_DKPTable, name, "player")
+	local path;
+
+	if search then
+		path = MonDKP_DKPTable[search[1][1]]
+		path.dkp = path.dkp + amount
+		path.lifetime_gained = path.lifetime_gained + amount;
+	end
+end
+
+function MonDKP:AwardRaid(includeRaid, includeStandby, amount, reason) -- includeStandby = True => Also awards to players on standby
+    local raidAwardList = ""; -- List of players in raid that are awarded DKP
+    local standbyAwardList = ""; -- List of players on standby that are awarded DKP
+    local curTime = time();
+    local curOfficer = UnitName("player")
+
+    if MonDKP:CheckRaidLeader() then -- only allows raid leader to disseminate DKP
```

Comment:
Disagree. This limits functionality. It should be officer from whitelist. RL doesn't need to be the one responsible for DKP. For us ML does it.

Reply:
For boss-kill DKP ("autoaward") this check is in place right now, see
https://github.com/Roeshambo/MonolithDKP/blob/0f5f77db5c19c8af192a0607f8d8da9cfb32cc19/Modules/AutoAward.lua

For time-based DKP (RaidTimer.lua), its also only given as raid-leader:
https://github.com/Roeshambo/MonolithDKP/blob/0f5f77db5c19c8af192a0607f8d8da9cfb32cc19/Modules/RaidTimer.lua#L119

So basically nothing is changed with this pull request in this regard. Why it is like that in the first place, i am not sure but @Roeshambo should know?

We could think about changing it in the future though. But it might create problems with keeping the DKP tables in sync if multiple officers assign DKP instead of one of them that then keeps syncing changes to them.